### ISSUE TITLE
Question about PrROIpooling

### ISSUE THREAD

**Issue by JoyHuYY1412** [vacancy/PreciseRoIPooling#28]
Hi, I am not so sure when do you use your `Algorithm 2` in your paper to refine the bounding box.

I think it works more like a finetune process, am I right?

So when I use PrROIPooling, the net work paramters should be fixed, and the features here
`roi_features = avg_pool(features, rois)` is detached form the network already?
then you set the requires_grad of features and rois to true to refine the rois, right?

**Comment by JoyHuYY1412** [comment]
Can it be used in training in order to train the box regression head?

**Comment by vacancy** [comment]
1. Yes - your interpretation is correct.
2. I am not sure what do you exactly mean by "Can it be used in training in order to train the box regression head?"

**Comment by JoyHuYY1412** [comment]
> 1. Yes - your interpretation is correct.
> 2. I am not sure what do you exactly mean by "Can it be used in training in order to train the box regression head?"

Like, in the finetuning process, first use RPN rois, then box regression, the use regressed-rois to calculate IOU, loss is the distance between IOU and 1, so you can use this loss to train regression head? But the feature cannot be fixed, only fix the IOU head in doing so.

**Comment by JoyHuYY1412** [comment]
BTW, in your current setting, is prROI only used in finetuning, not in training?

**Comment by vacancy** [comment]
I am not sure what do you exactly mean by "finetuning process". During INFERENCE, we use the IoU head to refine the bounding box iteratively, following algo2 in the paper.
